### PR TITLE
mandatory alternative receiver for gnosis safe app

### LINKED ISSUE #226: Alternative receiver must be mandatory if OB app is run within the Gnosis Safe
Since the OB UI can be used in the Gnosis Safe after #225, users must be forced to enter the tokens recipient address (the alternative receiver feature) as it is almost impossible to have the same Gnosis Safe address on both sides of the bridge. 

Consider to add a warning that a user should not send assets to the same address on the respective other network as their Safe likely does not exist there. Make adding a recipient address mandatory when the app is loaded within the Gnosis Safe context (can be checked with the Safe Provider), if the entered address equals the Safe address, a window with the operation confirmation must appear.

### PR DESCRIPTION
**Pull request by dan13ram**
Closes #226 

Screenshot from 2021-05-06 16:10:51

### CODE REVIEW

File: packages/react-app/src/components/warnings/GnosisSafeWarning.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <Flex align="center" direction="column" w="100%" mb="4">
+      <Alert
+        status="warning"
+        borderRadius={5}
+        boxShadow={noShadow ? 'none' : '0px 1rem 2rem rgba(204, 218, 238, 0.8)'}
+      >
+        <AlertIcon minWidth="20px" />
+        <Text fontSize="small">
+          It is mandatory to set an alternative recipient address when
```
Comment (akolotov) [inline packages/react-app/src/components/warnings/GnosisSafeWarning.jsx:18]: Consider to extend the message to say that usually the user running the OB UI as a Gnosis Safe App needs to specify the address of the Gnosis Safe on another side of the bridge.

**Review by akolotov — COMMENTED** [review]
Will these changes request for an additional confirmation if the user specifies the same address as the current Gnosis Safe wallet address?

File: packages/react-app/src/components/warnings/GnosisSafeWarning.jsx (hunk 1 above)
Comment (dan13ram) [reply packages/react-app/src/components/warnings/GnosisSafeWarning.jsx:18]: Sure. Can do!

**Comment by dan13ram** [comment]
> Will these changes request for an additional confirmation if the user specifies the same address as the current Gnosis Safe wallet address?

No it doesn't. It just does the usual confirmation screen. 

Do you think this should be an additional modal? Or we could have a check box somewhat similar to "Inflationary" tokens for the user to check everytime the UI is used from Gnosis Safe App.

**Comment by akolotov** [comment]
> Do you think this should be an additional modal? Or we could have a check box somewhat similar to "Inflationary" tokens for the user to check everytime the UI is used from Gnosis Safe App.

Well, the checkbox could be OK if it appears only for the case when a user specifies the same address in the alternative receiver field as the current Gnosis Safe wallet address.

**Comment by dan13ram** [comment]
@akolotov as requested I've added a checkbox for the user if they have specified the same gnosis safe address.

Bridge warning with extra sentence:
Screenshot from 2021-05-07 12:45:22

Confirmation modal for different receiver:
Screenshot from 2021-05-07 12:55:07

Confirmation modal for same receiver address:
Screenshot from 2021-05-07 12:51:31

**Review by maxaleks — APPROVED** [review]
LGTM

**Pull request merged**